### ISSUE TITLE
Faulx factory can not be used on parametrized if using xdist with multiple workers.

### ISSUE THREAD

**Issue by renzon** [omaciel/pytest-fauxfactory#15]
Xdists need the same test ids for all workers and since random data is generated, this is not respected. I got this pub while trying to run a test on Satellite 6:

https://github.com/pytest-dev/pytest-xdist/issues/149

We should be able to give the same random seed to all python parameters so the same set of data would be generated for each worker.

**Comment by renzon** [comment]
I have created a workaround on robottelo:

```python
def xdist_adapter(argvalues):
    """ Adapter to avoid error when running tests on xdist
    Check https://github.com/pytest-dev/pytest-xdist/issues/149

    Ig returns a dict with lst as argvalues and range(len(lst)) as ids

    Since every run has the same number of values, ids is going to be the same
    on different workers

    Ex:
        @pytest.mark.parametrize(
            'value', **xdist_adapter(invalid_boolean_strings()))
            @tier1
            def test_something(value):
                pass

    :param argvalues: to be passed to parametrize
    :return: dict
    """
    return {'argvalues': argvalues, 'ids': map(str, range(len(argvalues)))}
```

**Comment by omaciel** [comment]
Hmmm it would be tricky to get a similar adapter for the `faux_generator` method since we don't really know how many records would be returned... I guess that for `faux_string` and `faux_callable` we could use something similar to what you did for `robottelo` though.

**Comment by omaciel** [comment]
@renzon what do you think of #16 ?

**Comment by omaciel** [comment]
Solved by f1d14ee
